src/Elastic.Apm.MongoDb/MongoDbDiagnosticsSubscriber.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Diagnostics;
+using Elastic.Apm.DiagnosticSource;
+using Elastic.Apm.Mongo.DiagnosticSource;
+using Elastic.Apm;
+
+// ReSharper disable UnusedMember.Global
+
+namespace Elastic.Apm.Mongo
+{
+	/// <summary>
+	///     A subscriber to events from mongoDB driver diagnostic source.
+	/// </summary>
+	public class MongoDbDiagnosticsSubscriber : IDiagnosticsSubscriber
```

Comment:
We don't handle `enabled` and `recording` here. https://github.com/elastic/apm-agent-dotnet/pull/1161 is about to move that logic into a base class - I'll adapt this once https://github.com/elastic/apm-agent-dotnet/pull/1161 is ready to go.